app/uk/gov/hmrc/partnershipidentificationfrontend/views/helpers/CheckYourAnswersListBuilder.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.partnershipidentificationfrontend.views.helpers
+
+import play.api.i18n.Messages
+import play.api.mvc.Call
+import uk.gov.hmrc.govukfrontend.views.Aliases.{Actions, SummaryListRow, Value}
+import uk.gov.hmrc.govukfrontend.views.viewmodels.content.Text
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.{ActionItem, Key}
+import uk.gov.hmrc.partnershipidentificationfrontend.controllers.routes
+import uk.gov.hmrc.partnershipidentificationfrontend.models.PartnershipInformation
+
+import javax.inject.Singleton
+
+@Singleton
+class CheckYourAnswersListBuilder {
+
+  def build(journeyId: String,
+            partnershipInformation: PartnershipInformation)(implicit messages: Messages): Seq[SummaryListRow] =
+    maybeACompanyNumberRow(partnershipInformation, journeyId) ++
+      Seq(sautrRow(partnershipInformation, journeyId)) ++
+      maybeAPostcodeRow(partnershipInformation, journeyId)
+
+  private def maybeACompanyNumberRow(partnershipInformation: PartnershipInformation,
+                                     journeyId: String)(implicit messages: Messages): Seq[SummaryListRow] =
+    partnershipInformation.optCompanyProfile.
+      map(partnershipInformation =>
+        toSummaryListRow(
+          translationKeySuffix = "companyNumber",
+          valueToShow = partnershipInformation.companyNumber,
+          changeItCall = routes.CaptureCompanyNumberController.show(journeyId))
+      ).toSeq
```

Comment:
This one is `Seq[SummaryListRow]` and the maybeAPostcodeRow is `Option[SummaryListRow]`. Can these be consistent or is there a reason they're different?

Reply:
... that was a battle :) .... I wrote it as Option[SummaryListRow]  at first (because it is an optional  ) but Scala transform the return type of def build(..) into a Iterable[] if an option it is the first element of the return type instead of a Seq (that could be not a problem at all). We could leave it as Iterable .... as remove .toSeq  ... maybe we should pair because it to complicated to explain :-) :-) :-)